File: JINSEOP/프로그래머스_WEEK15/소수 만들기.cpp
#include <vector>
#include <string>
#include <cmath>
using namespace std;

int sum = 0, cnt = 0;
bool validate(int n) {
    for (int i = 2; i <= sqrt(n); i++)
        if (n % i == 0)
            return false;

    return true;
}

void backtrack(int depth, int idx, vector<int> nums, vector<bool>& visit) {
    if (depth == 3) {
        if (validate(sum))
            cnt++;

        return;
    }

    for (int i = idx; i < nums.size(); i++) {
        if (!visit[i]) {
            sum += nums[i];
            visit[i] = true;
            backtrack(depth + 1, i, nums, visit);
            sum -= nums[i];
            visit[i] = false;
        }
    }
}

int solution(vector<int> nums) {
    int answer = -1;
    
    vector<bool> visit(nums.size(), 0);
    backtrack(0, 0, nums, visit);

    answer = cnt;
    return answer;
}

File: DONGUK/프로그래머스_WEEK_15/소수 만들기.py
import math
from itertools import combinations

def isPrime(x):
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False 
        
    return True 


def solution(nums):
    answer = 0

    comb_list = list(combinations(nums, 3))
    
    ans_list = []
    
    for data in comb_list:
        if isPrime(sum(data)):
            answer += 1
            
    
    return answer